hail/python/hail/vds/combiner/variant_dataset_combiner.py

```diff
@@ -428,19 +428,11 @@ def step(self):
             self._job_id += 1
 
     def _write_final(self, vds):
-        fd = VariantDataset.ref_block_max_length_field
+        vds.write(self._output_path)
 
-        if fd not in vds.reference_data.globals:
+        if VariantDataset.ref_block_max_length_field not in vds.reference_data.globals:
             info("VDS combiner: computing reference block max length...")
-            max_len = vds.reference_data.aggregate_entries(
-                hl.agg.max(vds.reference_data.END + 1 - vds.reference_data.locus.position)
-            )
-            info(f"VDS combiner: max reference block length is {max_len}")
-            vds = VariantDataset(
-                reference_data=vds.reference_data.annotate_globals(**{fd: max_len}), variant_data=vds.variant_data
-            )
-
-        vds.write(self._output_path)
+            hl.vds.store_ref_block_max_length(self._output_path)
```

Comment:
I agree we shouldn't do this pass twice if not necessary, but is it not possible to compute this as we go so that we can just store this information in the globals? This feels like a hack that is pointing at some limitation of the Hail model.